### PR TITLE
Fix Tongue Based Speech Modifiers Being Applied To Users Native Languages

### PR DESCRIPTION
**Pull request by FearfulFurnishing**
## About The Pull Request
This change allows tongue based speech modifications to be ignored if the user is speaking in a native language or using hand signs, putting it back to where it was functionally before moving to speechmod components.
## Why It's Good For The Game
This is correcting some of the speaking code to how it was working prior to speechmods, meaning lizard people won't be elongating there s's in draconic. Fly people are the other species with a tongue based speech modifier and receive the same fix. This also corrects  tongue based speech mods getting applied to sign language. Speech modifiers are still applied if the user is talking in a non-native language, same as it was pre speechmod.

Before:
speechmod_demo_before
After:
speechmod_demo_after

## Changelog

:cl:
fix: fixing speech modifiers being applied to a tongue's native languages.
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
This should probably have a unit test

File: code/datums/components/speechmod.dm [hunk 1]
```diff
@@ -13,8 +13,10 @@
 	var/slots
 	/// If set to true, turns all text to uppercase
 	var/uppercase = FALSE
+	/// Any additional checks that we should do before applying the speech modification
+	var/datum/callback/should_modify_speech
```
Comment (SmArtKar) [inline code/datums/components/speechmod.dm:17]: You should null this on destroy, else you get a hanging ref

**Review by SmArtKar — CHANGES_REQUESTED** [review]

**Comment by FearfulFurnishing** [comment]
> This should probably have a unit test

Test case added now

File: code/datums/components/speechmod.dm [hunk 2]
```diff
@@ -51,6 +54,8 @@
 	var/message = speech_args[SPEECH_MESSAGE]
 	if(message[1] == "*")
 		return
+	if(!isnull(should_modify_speech) && !should_modify_speech?.Invoke(source, speech_args))
```
Comment (Ghommie) [inline code/datums/components/speechmod.dm:57]:
```suggestion
	if(!isnull(should_modify_speech) && !should_modify_speech.Invoke(source, speech_args))
```
should_modify_speech being null has already been checked earlier in the same conditional.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**